driver/gl/gl.go

```diff
@@ -172,113 +145,45 @@ func (c *glCanvas) newGlTextTexture(obj fyne.CanvasObject) uint32 {
 	height := scaleInt(c, img.Size().Height)
 	if width <= 0 || height <= 0 {
 		return 0
 	}
 
-	if img.File != "" || img.Resource != nil {
-		var file io.Reader
-		var name string
-		if img.Resource != nil {
-			name = img.Resource.Name()
-			file = bytes.NewReader(img.Resource.Content())
-		} else {
-			name = img.File
-			handle, _ := os.Open(img.File)
-			defer handle.Close()
-			file = handle
-		}
-
-		if strings.ToLower(filepath.Ext(name)) == ".svg" {
-			icon, err := oksvg.ReadIconStream(file)
-			if err != nil {
-				log.Println("SVG Load error:", err)
-
-				return 0
-			}
-			icon.SetTarget(0, 0, float64(width), float64(height))
-
-			w, h := int(icon.ViewBox.W), int(icon.ViewBox.H)
-			// this is used by our render code, so let's set it to the file aspect
-			img.PixelAspect = float32(w) / float32(h)
-			// if the image specifies it should be original size we need at least that many pixels on screen
-			if img.FillMode == canvas.ImageFillOriginal {
-				pixSize := fyne.NewSize(unscaleInt(c, w), unscaleInt(c, h))
-				img.SetMinSize(pixSize)
-			}
-
-			raw = image.NewRGBA(image.Rect(0, 0, width, height))
-			scanner := rasterx.NewScannerGV(w, h, raw, raw.Bounds())
-			raster := rasterx.NewDasher(width, height, scanner)
-
-			icon.Draw(raster, 1)
-		} else {
-			pixels, _, err := image.Decode(file)
-
-			if err != nil {
-				log.Println("image err", err)
-
-				return 0
-			}
-			origSize := pixels.Bounds().Size()
-			// this is used by our render code, so let's set it to the file aspect
-			img.PixelAspect = float32(origSize.X) / float32(origSize.Y)
-			// if the image specifies it should be original size we need at least that many pixels on screen
-			if img.FillMode == canvas.ImageFillOriginal {
-				pixSize := fyne.NewSize(unscaleInt(c, origSize.X), unscaleInt(c, origSize.Y))
-				img.SetMinSize(pixSize)
-			}
-
-			raw = image.NewRGBA(pixels.Bounds())
-			draw.Draw(raw, pixels.Bounds(), pixels, image.ZP, draw.Src)
-		}
-	} else if img.PixelColor != nil {
-		raw = image.NewRGBA(image.Rect(0, 0, width, height))
-		pixels := newPixelImage(img, c.Scale())
-
-		halfWidth := raw.Bounds().Size().X / 2
-		halfHeight := raw.Bounds().Size().Y / 2
+	if img.Raster != nil {
+		return c.img2Texture(img.Raster(width, height))
+	}
 
-		// use a WaitGroup so we don't return our image until it's complete
-		var wg sync.WaitGroup
-		wg.Add(4)
+	return c.img2Texture(image.NewRGBA(image.Rect(0, 0, 1, 1)))
 
-		go renderGlImagePortion(image.ZP, halfWidth, halfHeight, raw, pixels, &wg)
-		go renderGlImagePortion(image.Pt(0, halfHeight), halfWidth, height-halfHeight, raw, pixels, &wg)
-		go renderGlImagePortion(image.Pt(halfWidth, 0), width-halfWidth, halfHeight, raw, pixels, &wg)
-		go renderGlImagePortion(image.Pt(halfWidth, halfHeight), width-halfWidth, height-halfHeight, raw, pixels, &wg)
+}
 
-		wg.Wait()
-	} else {
-		raw = image.NewRGBA(image.Rect(0, 0, 1, 1))
+func (c *glCanvas) img2Texture(img image.Image) uint32 {
```

Comment:
Can you please change this to "imgToTexture" the number 2 is confusing (especially alongside the TexImage2D calls).